### PR TITLE
docs: list minimal permissions for Constellation setup

### PR DESCRIPTION
**Pull request by msanft**
### Proposed change(s)

- List the minimal permissions to be able to execute the `constellation iam` commands (`iam create` & `iam destroy`) for each CSP.

### Checklist

- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

File: docs/docs/workflows/config.md [hunk 1]
```diff
@@ -84,6 +84,13 @@ If you haven't generated a configuration file yet, you can do so by adding the `
 
 You must be authenticated with the [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) in the shell session.
 
+Your Azure account also needs to have at least the following permissions:
+* `Microsoft.Authorization/roleDefinitions/*`
+* `Microsoft.Authorization/roleAssignments/*`
+* `*/register/action`
```
Comment (Nirusu) [inline docs/docs/workflows/config.md:90]: I think this roughly needs to be in sync with the required subscription providers:
https://docs.edgeless.systems/constellation/getting-started/install#required-permissions

`*/register/action` can be reduced to the ones listed there under the condition that you run all Terraform commands with `ARM_SKIP_PROVIDER_REGISTRATION=true` (see the [troubleshooting](https://docs.edgeless.systems/constellation/workflows/troubleshooting#azure-resource-providers-cant-be-registered) link I recently added)

File: docs/docs/workflows/config.md [hunk 2]
```diff
@@ -84,6 +84,13 @@ If you haven't generated a configuration file yet, you can do so by adding the `
 
 You must be authenticated with the [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) in the shell session.
 
+Your Azure account also needs to have at least the following permissions:
```
Comment (thomasten) [inline docs/docs/workflows/config.md:87]:
```suggestion
You must be authenticated with the [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) in the shell session, and you need at least the following permissions:
```
nit: I'd shorten and link this to the previos sentence, but you can keep it if you prefer.

File: docs/docs/workflows/config.md (hunk 1 above)
Comment (msanft) [reply docs/docs/workflows/config.md:90]: > `*/register/action` can be omitted if you run all Terraform-related commands with `ARM_SKIP_PROVIDER_REGISTRATION=true`

Should I add this to the Azure text? Or would you completely replace it? And for "syncing" it with the resource providers listed at the installation step, I think the information there is incomplete? At least, if I try removing the `*/register/action` permission and run the command, I receive an error about at least 50 missing resource provider registrations from TF. (including the ones listed in your link though)

**Comment by katexochen** [comment]
@msanft Nice, thanks for the work! Could you please validate that we use that exact permissions in the CI for our e2e testing? That way we'll quickly notice if anything changes (at least if we require more permissions).

File: docs/docs/workflows/config.md (hunk 1 above)
Comment (Nirusu) [reply docs/docs/workflows/config.md:90]: Yes, this is why you need to set `ARM_SKIP_PROVIDER_REGISTRATION=true` in this case. 

The Azure Terraform provider has this slightly dumb behavior that it tries to register all resource providers it provides even if it never uses them. This makes it require way more permissions _by default_. But if you set `ARM_SKIP_PROVIDER_REGISTRATION=true` and register the actually required providers manually, the list of requires permissions is way lower (and if you manually register, you don't need `*/register/action`.

So given that this requires manual intervention, we could maybe put a disclaimer here saying that `*/register/action` is not needed if you set `ARM_SKIP_PROVIDER_REGISTRATION=true`, but you need to manually register the resource providers.

**Comment by msanft** [comment]
We should probably merge this with https://docs.edgeless.systems/constellation/getting-started/install#set-up-cloud-credentials and also extend the docs on which permissions are needed for cluster creation there. (as its already done on AWS)

File: docs/docs/getting-started/install.md [hunk 3]
```diff
@@ -119,24 +119,47 @@ The following [resource providers need to be registered](https://learn.microsoft
...
 By default, Constellation tries to register these automatically if they haven't been registered before.
 
-You need the following permissions for your user account:
-
-- `Contributor` (to create cloud resources)
-- `User Access Administrator` (to create a service account)
-
-If you don't have these permissions with scope *subscription*, ask your administrator to [create the service account and a resource group for your Constellation cluster](first-steps.md).
-Your user account needs the `Contributor` permission scoped to this resource group.
+To [create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:
```
Comment (Nirusu) [inline docs/docs/getting-started/install.md:122]:
```suggestion
To [automatically create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:
```
To highlight that this refers to our feature so we don't confuse people who want to set these up manually.

File: docs/docs/getting-started/install.md [hunk 4]
```diff
@@ -119,24 +119,47 @@ The following [resource providers need to be registered](https://learn.microsoft
...
+* `Microsoft.Authorization/roleDefinitions/*`
+* `Microsoft.Authorization/roleAssignments/*`
+* `*/register/action` (can be ommitted if the resource providers mentioned above are already registered and the `ARM_SKIP_PROVIDER_REGISTRATION` environment variable is set to `true` when creating the IAM configuration)
```
Comment (Nirusu) [inline docs/docs/getting-started/install.md:125]: I would make this a footnote maybe after and explain this after the list of all permissions. I think that may look cleaner in the docs eventually.

File: docs/docs/getting-started/install.md (hunk 3 above)
Comment (thomasten) [reply docs/docs/getting-started/install.md]: But doesn't manually creating it require the same permissions?

File: docs/docs/getting-started/install.md [hunk 5]
```diff
@@ -119,24 +119,102 @@ The following [resource providers need to be registered](https://learn.microsoft
...
-
-- `Contributor` (to create cloud resources)
-- `User Access Administrator` (to create a service account)
-
-If you don't have these permissions with scope *subscription*, ask your administrator to [create the service account and a resource group for your Constellation cluster](first-steps.md).
-Your user account needs the `Contributor` permission scoped to this resource group.
+To [automatically create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:
+* `Microsoft.Authorization/roleDefinitions/*`
+* `Microsoft.Authorization/roleAssignments/*`
+* `*/register/action` [1]
```
Comment (thomasten) [inline docs/docs/getting-started/install.md:125]:
```suggestion
* `*/register/action` \[1]
```
nit: better escape the opening bracket

File: docs/docs/getting-started/install.md [hunk 6]
```diff
@@ -119,24 +119,102 @@ The following [resource providers need to be registered](https://learn.microsoft
...
+* `Microsoft.Network/loadBalancers/*`
+* `Microsoft.Network/networkSecurityGroups/*`
+* `Microsoft.Network/loadBalancers/backendAddressPools/*`
+* `Microsoft.Network/virtualNetworks/subnets/*`
+* `Microsoft.Compute/virtualMachineScaleSets/*`
+* `Microsoft.ManagedIdentity/userAssignedIdentities/*`
+
+Follow Microsoft's guide on [understanding](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-definitions) and [assigning roles](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments).
+
+1: `*/register/Action` can be omitted if the resource providers mentioned above are already registered and the `ARM_SKIP_PROVIDER_REGISTRATION` environment variable is set to `true` when creating the IAM configuration
```
Comment (thomasten) [inline docs/docs/getting-started/install.md:142]:
```suggestion
1: You can omit `*/register/Action` if the resource providers mentioned above are already registered and the `ARM_SKIP_PROVIDER_REGISTRATION` environment variable is set to `true` when creating the IAM configuration.
```
nit: prefer active voice

File: docs/docs/getting-started/install.md [hunk 7]
```diff
@@ -145,7 +223,7 @@ Follow Google's guide on [understanding](https://cloud.google.com/iam/docs/under
 
 To set up a Constellation cluster, you need to perform two tasks that require permissions: create the infrastructure and create roles for cluster nodes. Both of these actions can be performed by different users, e.g., an administrator to create roles and a DevOps engineer to create the infrastructure.
 
-To create the AWS IAM policies, your user requires the following minimal set of permissions:
+To create the IAM configuration for Constellation, you need the following permissions:
```
Comment (thomasten) [inline docs/docs/getting-started/install.md:226]:
```suggestion
To [create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:
```

File: docs/docs/getting-started/install.md [hunk 8]
```diff
@@ -119,24 +119,102 @@ The following [resource providers need to be registered](https://learn.microsoft
...
+To [create a Constellation cluster](../workflows/create.md#the-create-step), you need the following permissions:
+* `Microsoft.Insights/components/*`
+* `Microsoft.Network/publicIPAddresses/*`
+* `Microsoft.Network/virtualNetworks/*`
+* `Microsoft.Network/loadBalancers/*`
+* `Microsoft.Network/networkSecurityGroups/*`
+* `Microsoft.Network/loadBalancers/backendAddressPools/*`
+* `Microsoft.Network/virtualNetworks/subnets/*`
+* `Microsoft.Compute/virtualMachineScaleSets/*`
+* `Microsoft.ManagedIdentity/userAssignedIdentities/*`
```
Comment (thomasten) [inline docs/docs/getting-started/install.md:138]:
```suggestion
* `Microsoft.Resources/subscriptions/resourcegroups/*`

The built-in `Owner` role is a superset of these permissions.

To [create a Constellation cluster](../workflows/create.md#the-create-step), you need the following permissions:
* `Microsoft.Insights/components/*`
* `Microsoft.Network/publicIPAddresses/*`
* `Microsoft.Network/virtualNetworks/*`
* `Microsoft.Network/loadBalancers/*`
* `Microsoft.Network/networkSecurityGroups/*`
* `Microsoft.Network/loadBalancers/backendAddressPools/*`
* `Microsoft.Network/virtualNetworks/subnets/*`
* `Microsoft.Compute/virtualMachineScaleSets/*`
* `Microsoft.ManagedIdentity/userAssignedIdentities/*`

The built-in `Contributor` role is a superset of these permissions.
```
Should we give the user simpler alternatives? (Although Owner is not that cool of course.)
If so, do we have something similar on GCP?

File: docs/docs/getting-started/install.md (hunk 3 above)
Comment (Nirusu) [reply docs/docs/getting-started/install.md]: True. Let’s just keep it as it is then 😄

File: docs/docs/getting-started/install.md (hunk 8 above)
Comment (msanft) [reply docs/docs/getting-started/install.md:138]: I think we can figure out the needed roles for GCP as well. But, as you already mentioned, they might be very extensive

**Comment by msanft** [comment]
`  {"message": "[Vale.Spelling] Did you really mean 'superset'?", "location": {"path": "docs/docs/getting-started/install.md", "range": {"start": {"line": 129, "column": 32}}}, "severity": "ERROR"}`

Any way to silence the review-🐶 here?

File: docs/docs/getting-started/install.md [hunk 9]
```diff
@@ -119,24 +119,110 @@ The following [resource providers need to be registered](https://learn.microsoft
 
 By default, Constellation tries to register these automatically if they haven't been registered before.
 
-You need the following permissions for your user account:
+To [automatically create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:
```
Comment (thomasten) [inline docs/docs/getting-started/install.md:122]:
```suggestion
To [create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:
```
Nils agreed to revert this https://github.com/edgelesssys/constellation/pull/1442#discussion_r1147527831

File: docs/docs/getting-started/install.md [hunk 10]
```diff
@@ -119,24 +119,110 @@ The following [resource providers need to be registered](https://learn.microsoft
...
+* `Microsoft.Authorization/roleDefinitions/*`
+* `Microsoft.Authorization/roleAssignments/*`
+* `*/register/action` \[1]
+* `Microsoft.ManagedIdentity/userAssignedIdentities/*`
+* `Microsoft.Resources/subscriptions/resourcegroups/*`
 
-- `Contributor` (to create cloud resources)
-- `User Access Administrator` (to create a service account)
+The built-in `Owner` role is a superset of these permissions.
```
Comment (thomasten) [inline docs/docs/getting-started/install.md:129]: > Any way to silence the review-🐶 here?

add it to docs/styles/Vocab/constellation/accept.txt

**Review by thomasten — APPROVED** [review]

File: docs/docs/getting-started/install.md [hunk 11]
```diff
@@ -138,14 +138,16 @@ To [create a Constellation cluster](../workflows/create.md#the-create-step), you
...
-* `Microsoft.Attestation/attestationProviders/*`
+* `Microsoft.Attestation/attestationProviders/*` \[2]
 
 The built-in `Contributor` role is a superset of these permissions.
 
 Follow Microsoft's guide on [understanding](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-definitions) and [assigning roles](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments).
 
 1: You can omit `*/register/Action` if the resource providers mentioned above are already registered and the `ARM_SKIP_PROVIDER_REGISTRATION` environment variable is set to `true` when creating the IAM configuration.
 
+2: You can omit `Microsoft.Attestation/attestationProviders/*` if `EnforceIDKeyDigest` is set to `MAAFallback` in the [config file](../workflows/config.md#configure-your-cluster).
```
Comment (Nirusu) [inline docs/docs/getting-started/install.md:149]:
```suggestion
2: You can omit `Microsoft.Attestation/attestationProviders/*` if `EnforceIDKeyDigest` is not set to `MAAFallback` in the [config file](../workflows/config.md#configure-your-cluster).
```

**Review by Nirusu — COMMENTED** [review]
Don't forget to set `Microsoft.Attestation` above for the required registered providers, too. You maybe can merge the footnotes for this, but not sure how it would look good given that they are in separate paragraphs.

**Pull request merged**